src/unwinder/libunwindstack_os.cpp

```diff
@@ -0,0 +1,42 @@
+extern "C" {
+#include <sentry.h>
+}
+
+#include <memory>
+#include <ucontext.h>
+#include <unwind.h>
+
+extern "C" {
+
+struct trace_state {
+    void **current;
+    void **end;
+};
+
+static _Unwind_Reason_Code
+unwind_callback(struct _Unwind_Context *context, void *arg)
+{
+    auto *state = static_cast<trace_state *>(arg);
+    uintptr_t pc = _Unwind_GetIP(context);
+    if (pc) {
+        if (state->current == state->end) {
+            return _URC_END_OF_STACK;
+        } else {
+            *state->current++ = reinterpret_cast<void *>(pc);
+        }
+    }
+    return _URC_NO_REASON;
+}
+
+__attribute__((visibility("default"))) size_t
+android_unwind_stack(
+    void *addr, const sentry_ucontext_t *uctx, void **ptrs, size_t max_frames)
+{
+    (void)addr;
+    (void)uctx;
+    trace_state state = { ptrs, ptrs + max_frames };
+    _Unwind_Backtrace(unwind_callback, &state);
```

Comment:
I always confuse the `_Unwind_*` and the `unw_*` APIs. So unfortunate that the first one does not allow unwinding from uctx.

Reply:
The `unw_` prefix is the "namespace" of `libunwind` which would be very convenient library and I would love to use, but AOSP dropped it and upstream doesn't really support Android.

But even if it would provide a `uctx` interface, I am not sure if the added complexity justifies the changes, as long as it doesn't offer any expected/significant improvements in the stack trace (for instance by mapping to `libunwindstack`).

This is especially true since most users use a pre-built package, which doesn't gain anything from the size improvements when using only a slim`mapper.so`, without ("manually") removing the `legacy.so`.

Reply:
`libunwind` is still [deployed](https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#unwinding) with NDK as the unwinder for C++ exception-handling and C scope-cleanups but it is not exposed as an app-developer library. 

The hint that NDK versions below r23 re-export the symbols of the previous libgcc unwinder, potentially screwing with client code that uses exception-handling, shows that stability of that "interface" is not a driver for decisions). Nonetheless, i can also try to access this implementation.